src/logic/nfts/rentals.ts

```diff
@@ -86,7 +86,10 @@ export function rentalNFTComponentShouldFetch(filters: NFTFilters): boolean {
     // The lookup for assets on rent is done by another component.
     !filters.isOnRent &&
     // Only Lands and Estates can be locked in the rentals contract.
-    !!filters.isLand &&
+    (!!filters.isLand ||
+      (!!filters.category &&
+        (filters.category === NFTCategory.ESTATE ||
+          filters.category === NFTCategory.PARCEL))) &&
```

Comment:
Would you mind adding some tests for this in the `src/tests/logic/nft-rentals.spec.ts` file?

Reply:
added!